### PR TITLE
Freeze `k8s.gcr.io` image registry

### PR DESCRIPTION
**Pull request by upodroid**
/cc @sftim @ameukam @kubernetes/release-engineering

I converted my [drafted email](https://docs.google.com/document/d/1-fAMIIZ9nLm4E6YJkRCYApFZZQl6RDaxMIlkt8EdR4I) to a blog post and copied the acknowledhements from https://kubernetes.io/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/

/label tide/merge-method-squash

### CODE REVIEW

File: content/en/blog/_posts/2023-02-05-k8s-gcr-io-is-frozen.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+layout: blog
+title: "k8s.gcr.io image registry will be frozen from the 1st of April 2023"
+date: 2023-02-05
+slug: k8s-gcr-io-is-frozen
```
Comment (sftim) [inline content/en/blog/_posts/2023-02-05-k8s-gcr-io-is-frozen.md:5]:
```suggestion
slug: k8s-gcr-freeze-announcement
```

File: content/en/blog/_posts/2023-02-05-k8s-gcr-io-is-frozen.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+**Authors**: Mahamed Ali (Rackspace Technology)
+
+The Kubernetes project has released a new community-owned image registry called `registry.k8s.io`. This registry has been generally available(GA) for several months, and we have published a [blog post]() about it, its benefits to the community, and the Kubernetes project. This post also announced that future versions of Kubernetes will not be available in the old registry.
```
Comment (sftim) [inline content/en/blog/_posts/2023-02-05-k8s-gcr-io-is-frozen.md:10]: FIXME - make this URL correct.

File: content/en/blog/_posts/2023-02-05-k8s-gcr-io-is-frozen.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+On the 1st of April 2023, the `k8s.gcr.io` image registry will be frozen and no further images for Kubernetes and related subprojects will be pushed to the old registry.
+
+What does this change mean for contributors:
+- If you are a maintainer of a subproject, you will need to update your manifests and helm charts to use the new registry.
```
Comment (sftim) [inline content/en/blog/_posts/2023-02-05-k8s-gcr-io-is-frozen.md:15]:
```suggestion
- If you are a maintainer of a subproject, you will need to update your manifests and Helm charts to use the new registry.
```

**Review by sftim — COMMENTED** [review]
Thanks

If we also want a banner at the top of the website for the 1st of April, we can do that (needs steering committee approval, and someone has to write the text).

Are we_sure_ about the date? The [first of April](https://en.wikipedia.org/wiki/April_Fools%27_Day) is the date most commonly associated with hoaxes. A banner that day might get overlooked.

**Comment by upodroid** [comment]
> If we also want a banner at the top of the website for the 1st of April, we can do that (needs steering committee approval, and someone has to write the text).

That would be nice, can you arrange that for me please?

I'll push the freeze date to 3rd of April 2023.

**Comment by BenTheElder** [comment]
I can help with the steering approval part, the banner text is in: 

https://github.com/kubernetes/website/blob/main/data/announcements/scheduled.yaml

The dockershim removal announcement should be an appropriate starting point, I think

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+title: "k8s.gcr.io image registry will be frozen from the 3rd of April 2023"
+date: 2023-02-06
+slug: k8s-gcr-io-freeze-announcement
+---
+
+**Authors**: Mahamed Ali (Rackspace Technology)
+
+The Kubernetes project has released a new community-owned image registry called `registry.k8s.io`. This registry has been generally available(GA) for several months, and we have published a [blog post](https://kubernetes.io/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) about it, its benefits to the community, and the Kubernetes project. This post also announced that future versions of Kubernetes will not be available in the old registry.
+
+On the 3rd of April 2023, the `k8s.gcr.io` image registry will be frozen and no further images for Kubernetes and related subprojects will be pushed to the old registry.
```
Comment (SergeyKanzhelev) [inline content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:12]: I'd suggest to start with what happens

```suggestion
The Kubernetes project runs a community-owned image registry called `registry.k8s.io` to host it's container images. On the 3rd of April 2023, the old registry `k8s.gcr.io` will be frozen and no further images for Kubernetes and related subprojects will be pushed to the old registry.

This registry `registry.k8s.io` replaced the old one and has been generally available (GA) for several months. We have published a [blog post](/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) about its benefits to community and the Kubernetes project. This post also announced that future versions of Kubernetes will not be available in the old registry. Now that time has come.
```

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+What does this change mean for contributors:
+- If you are a maintainer of a subproject, you will need to update your manifests and Helm charts to use the new registry.
+
+What does this change mean for end users:
+- 1.27 Kubernetes release will not be published to the old registry.
```
Comment (SergeyKanzhelev) [inline content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:18]: also patches for older versions of k8s I assume? Let's mention them here

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+- Starting in 1.25, the default image registry has been set to `registry.k8s.io`. This value is overridable but will no longer work for any version of Kubernetes released after April as they won’t be present in the old registry.
```
Comment (SergeyKanzhelev) [inline content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:19]: > This value is overridable but will no longer work for any version of Kubernetes

this sentence is confusing. Overriding it is desirable (to some private registry controlled by user or cloud provider) and will work for all versions. Using the old value will not work.

Please rephrase

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+- If you are running Kubernetes in networks where outbound traffic is restricted, you should consider hosting local image registry mirrors.
```
Comment (SergeyKanzhelev) [inline content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:20]:
```suggestion
- If you want to increase reliability of your cluster and remove dependency on community-owned registry or you are running Kubernetes in networks where external traffic is restricted, you should consider hosting local image registry mirrors. Some cloud vendors may offer hosted solutions for this.
```

Comment (SergeyKanzhelev) [reply content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:20]: let's emphasize that reliability will increase by removing dependency on community owned registry

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+Timeline of the Changes:
+
+- `k8s.gcr.io` will be frozen on the 3rd of April 2023
+- 1.27 is expected to be released on the 12th of April 2023
+- The last 1.23 release on `k8s.gcr.io` will be 1.23.18 (1.23 goes EoL before the freeze)
+- The last 1.24 release on `k8s.gcr.io` will be 1.24.12
+- The last 1.25 release on `k8s.gcr.io` will be 1.25.8
+- The last 1.26 release on `k8s.gcr.io` will be 1.26.3
+
+
```
Comment (SergeyKanzhelev) [inline content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:31]: Can you add a section either here or in documents that will offer a simple script to iterate thru images of all Pods and warn if any of them are using old registry?

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md (hunk 6 above)
Comment (upodroid) [reply content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:19]: I meant overriding the registry with k8s.gcr.io won't be useful after April. I'll rephrase.

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md (hunk 4 above)
Comment (upodroid) [reply content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:12]: ack, copied over to the email doc.

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md (hunk 7 above)
Comment (upodroid) [reply content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:20]: ack, copied over to the email doc.

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+## Timeline of the Changes:
+
+- `k8s.gcr.io` will be frozen on the 3rd of April 2023
+- 1.27 is expected to be released on the 12th of April 2023
+- The last 1.23 release on `k8s.gcr.io` will be 1.23.18 (1.23 goes EoL before the freeze)
+- The last 1.24 release on `k8s.gcr.io` will be 1.24.12
+- The last 1.25 release on `k8s.gcr.io` will be 1.25.8
+- The last 1.26 release on `k8s.gcr.io` will be 1.26.3
+
+You can run this command to list the images used by pods:
```
Comment (SergeyKanzhelev) [inline content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:32]:
```suggestion

## What's next

Please make sure your cluster does not have dependencies on old image registry. For example,  you can run this command to list the images used by pods:
```

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+```shell
+kubectl get pods --all-namespaces -o jsonpath="{.items[*].spec.containers[*].image}" |\
+tr -s '[[:space:]]' '\n' |\
+sort |\
+uniq -c
+```
+
```
Comment (SergeyKanzhelev) [inline content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:40]:
```suggestion

There may be other dependencies to old image registry. Make sure to review any potential dependencies to keep your cluster healthy and up to date.
```

**Comment by reylejano** [comment]
> Thanks
> 
> If we also want a banner at the top of the website for the 1st of April, we can do that (needs steering committee approval, and someone has to write the text).
> 
> Are we_sure_ about the date? The [first of April](https://en.wikipedia.org/wiki/April_Fools%27_Day) is the date most commonly associated with hoaxes. A banner that day might get overlooked.

Imo, we should avoid any announcements or new banners on the 1st of April

**Comment by upodroid** [comment]
The change will take effect on the 3rd of April now. I'll raise a separate PR for the banner.

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
+---
+layout: blog
+title: "k8s.gcr.io image registry will be frozen from the 3rd of April 2023"
+date: 2023-02-06
```
Comment (sftim) [inline content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:4]:
```suggestion
date: 2023-02-06T17:00:00
```

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+What does this change mean for contributors:
+- If you are a maintainer of a subproject, you will need to update your manifests and Helm charts to use the new registry.
+
+What does this change mean for end users:
+- 1.27 Kubernetes release will not be published to the old registry.
+- Patch releases for 1.24, 1.25, and 1.26 will no longer be published to the old registry from April. Please read the timelines below for details of the final patch releases in the old registry.
+- Starting in 1.25, the default image registry has been set to `registry.k8s.io`. This value is overridable in `kubeadm` and `kubelet` but setting it to `k8s.gcr.io` will fail for new releases after April as they won’t be present in the old registry.
+- If you want to increase the reliability of your cluster and remove dependency on the community-owned registry or you are running Kubernetes in networks where external traffic is restricted, you should consider hosting local image registry mirrors. Some cloud vendors may offer hosted solutions for this.
+
+## Timeline of the Changes:
```
Comment (sftim) [inline content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:23]:
```suggestion
## Timeline of the changes
```

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+- `k8s.gcr.io` will be frozen on the 3rd of April 2023
+- 1.27 is expected to be released on the 12th of April 2023
+- The last 1.23 release on `k8s.gcr.io` will be 1.23.18 (1.23 goes EoL before the freeze)
```
Comment (sftim) [inline content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:27]:
```suggestion
- The last 1.23 release on `k8s.gcr.io` will be 1.23.18 (1.23 goes end-of-life before the freeze)
```

**Review by sftim — COMMENTED** [review]
/lgtm
with nits

/approve
/hold

We can publish this as is. Even better with fixups first.

@upodroid you can unhold this at the time you need to.

File: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md (hunk 13 above)
Comment (upodroid) [reply content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md:27]: applied all the suggestions

**Comment by upodroid** [comment]
:( forgot to push

**Pull request merged**